lru.go

```diff
@@ -38,10 +39,10 @@ func New(size int) (*Cache, error) {
 
 // Purge is used to completely clear the cache
 func (c *Cache) Purge() {
-	c.lock.Lock()
-	defer c.lock.Unlock()
-	c.evictList = list.New()
-	c.items = make(map[interface{}]*list.Element, c.size)
+	// Evict each element in the list.
```

Comment:
Probably faster to just grab the lock once and iterate over the map (calling OnEvict if necessary). Otherwise this is doing a lot of lock acquisitions (one for len and one for remove oldest) per element.